src/MongoDB/ReadConcern.c

```diff
@@ -97,17 +97,49 @@ static PHP_METHOD(ReadConcern, isDefault)
 	RETURN_BOOL(mongoc_read_concern_is_default(intern->read_concern));
 } /* }}} */
 
+static HashTable* php_phongo_read_concern_get_properties_hash(zval* object, bool is_debug TSRMLS_DC) /* {{{ */
+{
+	php_phongo_readconcern_t* intern;
+	HashTable*                props;
+	const char*               level;
+
+	intern = Z_READCONCERN_OBJ_P(object);
+
+	PHONGO_GET_PROPERTY_HASH_INIT_PROPS(is_debug, intern, props, 1);
+
+	if (!intern->read_concern) {
+		return props;
+	}
+
+	level = mongoc_read_concern_get_level(intern->read_concern);
+
+	if (level) {
+#if PHP_VERSION_ID >= 70000
+		zval z_level;
+
+		ZVAL_STRING(&z_level, level);
+		zend_hash_str_update(props, "level", sizeof("level") - 1, &z_level);
+#else
+		zval* z_level;
+
+		MAKE_STD_ZVAL(z_level);
+		ZVAL_STRING(z_level, level, 1);
+		zend_hash_update(props, "level", sizeof("level"), &z_level, sizeof(z_level), NULL);
+#endif
+	}
+
+	return props;
+} /* }}} */
+
 /* {{{ proto array MongoDB\Driver\ReadConcern::bsonSerialize()
 */
 static PHP_METHOD(ReadConcern, bsonSerialize)
 {
-	const mongoc_read_concern_t* read_concern = phongo_read_concern_from_zval(getThis() TSRMLS_CC);
-
 	if (zend_parse_parameters_none() == FAILURE) {
 		return;
 	}
 
-	php_phongo_read_concern_to_zval(return_value, read_concern);
+	ZVAL_ARR(return_value, php_phongo_read_concern_get_properties_hash(getThis(), true TSRMLS_CC));
```

Comment:
Do I understand correctly that you'd never want to use `ZVAL_ARR()` with a non-temp properties hash? It looks like the macro just sucks the HashTable into the zval, so there's no added reference or anything.

Reply:
Yes, I believe so. It's a bit of an odd one out here.